internal/telemetry/client_test.go

```diff
@@ -13,60 +13,38 @@ import (
 	"reflect"
 	"sort"
 	"sync"
-	"sync/atomic"
 	"testing"
 	"time"
 
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
 )
 
 func TestClient(t *testing.T) {
-	ch := make(chan telemetry.RequestType)
-	var (
-		hb           sync.WaitGroup
-		gotheartbeat int64
-	)
-	hb.Add(1) // signal that we got a heartbeat
+	heartbeat := make(chan struct{})
```

Comment:
NIT: I usually use this to reduce the chances of leaking a goroutine that's trying to send on the channel if the reader goroutine exists unexpectedly. This does not seem to be an issue with this particular test, so feel free to ignore.

```suggestion
	heartbeat := make(chan struct{}, 1)
```

Reply:
Yes, I opted for a non-blocking send in the server handler for the same reason of avoiding a leak. 👍

Reply:
Yeah, I saw that – thanks for thinking about these edge cases! The only advantage of (ab)using the channel capacity for this is that it's less code 🙈